src/source/PeerConnection/SessionDescription.c

```diff
@@ -582,6 +610,42 @@ STATUS populateSessionDescriptionDataChannel(PKvsPeerConnection pKvsPeerConnecti
     return retStatus;
 }
 
+BOOL isPresentInRemote(PKvsRtpTransceiver pKvsRtpTransceiver, PSessionDescription pRemoteSessionDescription) {
+
+    PCHAR remoteAttributeValue, end;
+    UINT32 remoteTokenLen;
+
+    MEDIA_STREAM_TRACK_KIND localTrackKind = pKvsRtpTransceiver->sender.track.kind;
+
+    for (int i = 0; i < pRemoteSessionDescription->mediaCount; i++) {
+        PSdpMediaDescription pRemoteMediaDescription = &pRemoteSessionDescription->mediaDescriptions[i];
+        remoteAttributeValue = pRemoteMediaDescription->mediaName;
+        
+        if ((end = STRCHR(remoteAttributeValue, ' ')) != NULL) {
+            remoteTokenLen = (end - remoteAttributeValue);
+        } else {
+            remoteTokenLen = STRLEN(remoteAttributeValue);
+        }
+
+        switch (localTrackKind) {
+            case MEDIA_STREAM_TRACK_KIND_AUDIO:
+                if ( remoteTokenLen == STRLEN(MEDIA_SECTION_AUDIO_VALUE) && STRNCMP(MEDIA_SECTION_AUDIO_VALUE, remoteAttributeValue, remoteTokenLen) == 0 ) {
+                    return TRUE;
+                }
+                break;
+            case MEDIA_STREAM_TRACK_KIND_VIDEO:
+                if ( remoteTokenLen == STRLEN(MEDIA_SECTION_VIDEO_VALUE) && STRNCMP(MEDIA_SECTION_VIDEO_VALUE, remoteAttributeValue, remoteTokenLen) == 0 ) {
+                    return TRUE;
+                }
+                break;
+            default:
+                return FALSE;
+        }
+    }
+
+    return FALSE;
```

Comment:
I recommend functions having a single exit. I would have defined a boolean which will be set to FALSE by default and will only be reset in case of the audio/video hit and will only be returned here